scheduler/test/cook/test/kubernetes/controller.clj

```diff
@@ -24,6 +26,57 @@
+                  controller/log-weird-state (fn [_ _] :illegal)
+                  controller/pod-has-just-completed (fn [_] {:cook-expected-state :cook-expected-state/completed})
+                  controller/write-status-to-datomic (fn [_] :illegal)]
+
+      (is (= nil) (do-process :cook-expected-state/completed :missing))
+      (is (= nil) (do-process :cook-expected-state/completed :pod/succeeded))
+      (is (= :cook-expected-state/completed) (do-process :cook-expected-state/completed :pod/failed))
+      (is (= :cook-expected-state/completed) (do-process :cook-expected-state/completed :pod/running))
+      (is (= :cook-expected-state/completed) (do-process :cook-expected-state/completed :pod/unknown))
+      (is (= :cook-expected-state/completed) (do-process :cook-expected-state/completed :pod/waiting))
+
+      (is (= :cook-expected-state/killed) (do-process :cook-expected-state/killed :missing))
+      (is (= :cook-expected-state/completed) (do-process :cook-expected-state/killed :pod/succeeded))
+      (is (= :cook-expected-state/completed) (do-process :cook-expected-state/killed :pod/failed))
+      (is (= :cook-expected-state/killed) (do-process :cook-expected-state/killed :pod/running))
+      (is (= :cook-expected-state/killed) (do-process :cook-expected-state/killed :pod/unknown))
+      (is (= :cook-expected-state/killed) (do-process :cook-expected-state/killed :pod/waiting))
+
+      (is (= :cook-expected-state/completed) (do-process :cook-expected-state/running :missing))
+      (is (= :cook-expected-state/completed) (do-process :cook-expected-state/running :pod/succeeded))
+      (is (= :cook-expected-state/completed) (do-process :cook-expected-state/running :pod/failed))
+      (is (= :cook-expected-state/running) (do-process :cook-expected-state/running :pod/running))
+      (is (= :cook-expected-state/completed) (do-process :cook-expected-state/running :pod/unknown))
+      (is (= :cook-expected-state/completed) (do-process :cook-expected-state/running :pod/waiting))
+
+      (is (= :cook-expected-state/starting) (do-process :cook-expected-state/starting :missing))
+      (is (= :cook-expected-state/completed) (do-process :cook-expected-state/starting :pod/succeeded))
+      (is (= :cook-expected-state/completed) (do-process :cook-expected-state/starting :pod/failed))
+      (is (= :cook-expected-state/running) (do-process :cook-expected-state/starting :pod/running))
+      (is (= :cook-expected-state/completed) (do-process :cook-expected-state/starting :pod/unknown))
+      (is (= :cook-expected-state/starting) (do-process :cook-expected-state/starting :pod/waiting))
+
+      (is (= nil) (do-process :missing :missing))
+      (is (= nil) (do-process :missing :pod/succeeded))
+      (is (= nil) (do-process :missing :pod/failed))
+      (is (= nil) (do-process :missing :pod/running))
+      (is (= nil) (do-process :missing :pod/unknown))
+      (is (= nil) (do-process :missing :pod/waiting)))))
```

Comment:
Seems like all of these have the function call outside of the `(= ...)` form instead of inside

Reply:
Also, please use `nil?` instead of `(= nil ...)`

Reply:
Good find. Thanks.